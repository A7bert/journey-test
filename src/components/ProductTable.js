import React from 'react'
import Table from 'react-bootstrap/Table'

export default function ProductTable({currency, products}) {
  return (
    <div data-testid="table">
      <Table striped hover>
        <thead>
          <tr>
            <td colSpan='3'>Products ({currency}) </td>
          </tr>
          <tr>
            <td>Name</td>
            <td>Price</td>
            <td>Type</td>
          </tr>
        </thead>
        <tbody>
          {
            // ideally would use UUID for product id, but the id in the data is sequential
            products.map(product => {
              return (
                <tr key={`${product.id}-${product.type}`}>
                  <td>{product.name}</td>
                  <td>{product.price}</td>
                  <td>{product.type}</td>
                </tr>
              )
            })
          }
        </tbody>
      </Table>
    </div>
  )
}
