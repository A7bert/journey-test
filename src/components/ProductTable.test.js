import React from 'react'
import ReactDOM from 'react-dom'
import ProductTable from './ProductTable'
import LawnmowerRepository from '../repositories/LawnmowerRepository'

import renderer from 'react-test-renderer'

it("renders without crashing", () => {
  const div = document.createElement("div");
  const products = new LawnmowerRepository().getAll();
  ReactDOM.render(<ProductTable products={products} currency="NZD"></ProductTable>, div);
})

it("matches snapshot", () => {
  const products = new LawnmowerRepository().getAll();
  const tree = renderer.create(<ProductTable products={products} />).toJSON()
  expect(tree).toMatchSnapshot();
})
