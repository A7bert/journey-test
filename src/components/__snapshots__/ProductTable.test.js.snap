// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="table"
>
  <table
    className="table table-striped table-hover"
  >
    <thead>
      <tr>
        <td
          colSpan="3"
        >
          Products (
          ) 
        </td>
      </tr>
      <tr>
        <td>
          Name
        </td>
        <td>
          Price
        </td>
        <td>
          Type
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Hewlett-Packard Rideable Lawnmower
        </td>
        <td>
          3000
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Fisher Price's My First Lawnmower
        </td>
        <td>
          45
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Volkswagen LawnMaster 39000B Lawnmower
        </td>
        <td>
          1020
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
